src/include/api.h

```diff
@@ -111,26 +111,21 @@
                   (s), WT_CONFIG_REF(s, struct_name##_##func_name), (config), 0));          \
         }
 
-#define API_END(s, ret)                                                                    \
-    if ((s) != NULL) {                                                                     \
-        WT_TRACK_OP_END(s);                                                                \
-        WT_SINGLE_THREAD_CHECK_STOP(s);                                                    \
-        if ((ret) != 0 && __set_err)                                                       \
-            __wt_txn_err_set(s, (ret));                                                    \
-        if ((s)->api_call_counter == 1 && !F_ISSET(s, WT_SESSION_INTERNAL))                \
-            __wt_op_timer_stop(s);                                                         \
-        /*                                                                                 \
-         * We should not leave any history store cursor open when return from an api call. \
-         * However, we cannot do a stricter check before WT-7247 is resolved.              \
```

Comment:
Instead of deleting this check, is it possible to add to session a count of cached history store cursors and use that to fix this check to work with history store cursors cached?

Reply:
For this check, I don't see a way to fix it as there is no limit how many history store cursors can be open at the time now.